batch/batch/driver/instance_pool.py

```diff
@@ -129,6 +70,54 @@ def __init__(self, app, machine_name_prefix):
 
         self.name_instance = {}
 
+    async def update_zones(self):
+        northamerica_regions = {
+            # 'northamerica-northeast1',
+            'us-central1',
+            'us-east1',
+            'us-east4',
+            'us-west1',
+            'us-west2',
+            'us-west3',
+            'us-west4'
+        }
+
+        new_zones = []
+        new_zone_weights = []
+
+        async for r in await self.compute_client.list('/regions'):
+            name = r['name']
+            if name not in northamerica_regions:
+                continue
+
+            quota_remaining = {
+                q['metric']: q['limit'] - q['usage']
+                for q in r['quotas']
+            }
+
+            remaining = quota_remaining['PREEMPTIBLE_CPUS'] // self.worker_cores
+            if self.worker_local_ssd_data_disk:
+                remaining = min(remaining, quota_remaining['LOCAL_SSD_TOTAL_GB'] // 350)
+            else:
+                remaining = min(remaining, quota_remaining['SSD_TOTAL_GB'] // self.worker_pd_ssd_data_disk_size_gb)
+
+            weight = max(remaining // len(r['zones']), 1)
```

Comment:
This is basically either 0 or 1? It looks like you don't weight by the amount remaining.

Reply:
That's a `max`.  I always make this mistake, too.  So if remaining = 300 and there are three zones, the weight is max(300 // 3, 1) = 100 per zone.